### PR TITLE
Polygonal Surface Reconstruction + SCIP dependency added

### PR DESCRIPTION
**Pull request by petrasvestartas**
Polygonal Surface Reconstruction was added.

SCIP dependency was added and checked on windows only, wondering if this dependency will work on MAC too. 
I guess the name of scip, on mac will be "scip" instead of windows "libscip".

The polygonal surface reconstruction involves steps:
a) pointcloud normal estimation
b) plane creation and cropping to an average bounding box
c) optimal detection of suitable planes for the closed surface (SCIP) dependency.
result

cgal_polygonal_surface_reconstruction_ransac

The more I work on this, it feels that more methods for pointcloud processing would be useful to add, as well bringing CGAL C++ version to the current one.

### CODE REVIEW

**Comment by tomvanmele** [comment]
you have to change `setup.py`...

`libraries=["mpfr", "gmp", "libscip"]` => `libraries=["mpfr", "gmp", "scip"]`

File: docs/examples/polygonal_poinset_reconstruction.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+# Create polygons from vertices and faces
+polygons = [Polygon([Point(*vertices[i]) for i in face]) for face in faces]
+
+# Create a polyhedron from vertices and faces
+shape = Polyhedron(vertices.tolist(), faces.tolist())  # revise the Shape API
+
+# Create a viewer
+viewer = App(width=1600, height=900)
+viewer.view.camera.position = [5, -4, 2]
+viewer.view.camera.look_at([0, 1, 0])
```
Comment (tomvanmele) [inline docs/examples/polygonal_poinset_reconstruction.py:28]: i would move this to the end, into a "visualization" block. this makes it clearer where the algorithm part is and where the viz...

File: setup.py [hunk 2]
```diff
@@ -60,11 +60,12 @@ def get_library_dirs():
                 "src/triangulations.cpp",
                 "src/skeletonization.cpp",
                 "src/reconstruction.cpp",
+                "src/polygonal_surface_reconstruction.cpp",
             ]
         ),
         include_dirs=["./include", get_eigen_include(), get_pybind_include()],
         library_dirs=[get_library_dirs()],
-        libraries=["mpfr", "gmp"],
+        libraries=["mpfr", "gmp", "libscip"],
```
Comment (tomvanmele) [inline setup.py:68]: `libscip` => `scip`

File: src/compas_cgal/polygonal_surface_reconstruction.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+from nptyping import NDArray, Shape, Float
+import numpy as np
+from compas.geometry import Point, Vector
+from compas_cgal._cgal import polygonal_surface_reconstruction
+
+
+def polygonal_surface_reconstruction_ransac(
+    points: Union[list[Point], NDArray[Shape["Any, 3"], Float]],
+    normals: Union[list[Vector], NDArray[Shape["Any, 3"], Float]],
+):  # -> Tuple[NDArray[Shape["Any, 3"], Float], NDArray[Shape["Any, 3"], Int]]:
```
Comment (tomvanmele) [inline src/compas_cgal/polygonal_surface_reconstruction.py:12]: no need to use type comments. you can just use proper annotations...

File: setup.py (hunk 2 above)
Comment (petrasvestartas) [reply setup.py:68]: On Windows this results in:
`LINK : fatal error LNK1181: cannot open input file 'scip.lib'`

Even though it is installed through conda:
image


Can I specify like this:

```python
def get_scip_library():
    if windows:
        return "libscip"
    return "scip"

ext_modules = [

    Extension(
        "compas_cgal._cgal",
        sorted(
            [
                "src/compas_cgal.cpp",
                "src/compas.cpp",
                "src/meshing.cpp",
                "src/booleans.cpp",
                "src/slicer.cpp",
                "src/intersections.cpp",
                "src/measure.cpp",
                "src/subdivision.cpp",
                "src/triangulations.cpp",
                "src/skeletonization.cpp",
                "src/reconstruction.cpp",
                "src/polygonal_surface_reconstruction.cpp",
            ]
        ),
        include_dirs=["./include", get_eigen_include(), get_pybind_include()],
        library_dirs=[get_library_dirs()],
        libraries=["mpfr", "gmp", get_scip_library()],
        language="c++",
    ),
]


```

**Review by tomvanmele — APPROVED** [review]

**Pull request merged**